### PR TITLE
Add support for parallel execution of readonly functions

### PR DESCRIPTION
**Pull request by daniel-chambers**
JIRA: [NDC-244](https://hasurahq.atlassian.net/browse/NDC-244)

This PR adds support for executing query requests with variables in a parallel fashion against the underlying function.

For more information, please see the changes to the readme.

[NDC-244]: https://hasurahq.atlassian.net/browse/NDC-244?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -136,18 +147,24 @@ These types are unsupported as function parameter types or return types for func
...
-  return x + y;
+### Parallel execution
+If functions are involved remote relationships in your Hasura metadata, then they may be queried in a [batch-based fashion](https://hasura.github.io/ndc-spec/specification/queries/variables.html). In this situation, any async functions that are marked with the `@readonly` JSDoc tag may be executed in parallel. The default degree of parallelism per query request to the connector is 10, but you may customise this by using the `@paralleldegree` JSDoc tag on your function.
+
+``` typescript
+/**
+ * This function will only run up to 5 http requests in parallel per query
+ *
+ * @readonly
+ * @paralleldegree 5
```
Comment (sordina) [inline README.md:158]: Could you use params to an annotation such as `@parallel(degree: 5)`? No worries if not.

File: ndc-lambda-sdk/src/util.ts [hunk 2]
```diff
@@ -29,3 +29,7 @@ export function getFlags(flagsEnum: Record<string, string | number>, value: numb
           : []
     });
 }
+
+export function sleep(ms: number): Promise<void> {
+  return new Promise((r) => setTimeout(r, ms))
+}
```
Comment (sordina) [inline ndc-lambda-sdk/src/util.ts:35]: What's this used for again?

**Review by sordina — APPROVED** [review]
Hey @daniel-chambers 

This looks great. Wondering about the annotation name, could it have a parameter for degree instead? Anyway, happy if this is what you'd prefer.

File: ndc-lambda-sdk/src/util.ts (hunk 2 above)
Comment (daniel-chambers) [reply ndc-lambda-sdk/src/util.ts:35]: It's used in the tests to stagger function execution to test that parallelism is actually happening.

File: README.md (hunk 1 above)
Comment (daniel-chambers) [reply README.md:158]: Is that a thing in JSDoc? I've never seen that syntax before... what's something that does that already that I can use for reference?

File: ndc-lambda-sdk/src/util.ts (hunk 2 above)
Comment (sordina) [reply ndc-lambda-sdk/src/util.ts:35]: 👍

File: README.md (hunk 1 above)
Comment (sordina) [reply README.md:158]: I guess it's not a thing. There is the `@ann {bla}` convention, but we can just keep it as is.

Comment (daniel-chambers) [reply README.md:158]: That's too bad, I liked your syntax 😟

**Pull request merged**